### ISSUE TITLE
Unix socket compatibility problem

### ISSUE THREAD

**Issue by ChrisWang2014** [Microsoft/Docker.DotNet#118]
//case "unix":
                // TODO

May i ask when dockerClient is going to support unix://var/run/docker.sock as this is supported in other docker remote api client such as python one.

**Comment by stefanprodan** [comment]
Hello,

This has been implemented [here](https://github.com/Microsoft/Docker.DotNet/blob/master/Docker.DotNet/DockerClient.cs#L115).

 I've tested it by running inside a container with docker socket mounted as a volume. There is no NuGet package with the unix socket implementation, you need to build from source.

**Comment by ChrisWang2014** [comment]
Thanks for the prompt response. 
I am using docker for windows from which the daemon is at tcp://127.0.0.1:2375, the dockerclient is situated within one of the mono container that tries to shut down the other containers.

The endpoint for the client i am using is 
`_dockerClientEndpoint = isUnix ? "unix://var/run/docker.sock" : "tcp://127.0.0.1:2375";`

This is the daemon command im trying at the moment:
`docker run --rm -it --link databus --link database --link mailhog --link mailer --net mailer2integrationtest_default --privileged tester -H tcp://127.0.0.1:2375`

Appreantly i am getting System.Net.Http.HttpRequestException : Connection failed

Any fix or method for which a "master container" can work in windows environment (docker for windows)

Really appreciate any help, thanks

**Comment by stefanprodan** [comment]
You have to mount the socket inside the container to access `unix://var/run/docker.sock`
```
-v /var/run/docker.sock:/var/run/docker.sock
```

**Comment by ChrisWang2014** [comment]
yeah with the volume switch and env host name that got fixed, but now i am getting this 
`_wapi_connect: error looking up socket handle 0x7
    Mailer2IntegrationTest.IntegrationTest.TestSendsSidelinedMessageWhenSmtpServerBecomesAvailable [FAIL]
      System.IO.IOException : Unable to read data from the transport connection: Connection reset by peer.
      ---- System.Net.Sockets.SocketException : Connection reset by peer
      Stack Trace:`

**Comment by smartsingam** [comment]
Error:
No unix socket support on windows

**Comment by galvesribeiro** [comment]
Guys, there is no way to use Unix Socket mode on Windows (Docker4Windows or Windows Server Containers). There is no docker .socket file on Windows so you can't mount yet the host daemon and access it from a container. 

What you can do, is open the Windows daemon to listen to TCP connections, and from inside the container use the container network IP address of the host (usually the gateway IP assigned to the container) as the endpoint address. It should work.

**Comment by smartsingam** [comment]
can you explain how to connect mongodb live server using robomongo

**Comment by shairozan** [comment]
There may be no unix socket support, but for those of us writing in DotNetCore, we can do this in Linux. This is kind of a step back, since I can communicate with the Docker Socket in most other consuming models of the API

**Comment by kukjevov** [comment]
Is this still problem, because as i saw source code there is support for unix sockets and i have tested it yesterday and it worked.

**Comment by jterry75** [comment]
@kukjevov - Unix sockets are supported in the library. But only on Linux hosts. I believe what others want is to use Unix sockets on Windows hosts which is not possible at present

**Comment by caelchristian** [comment]
For those on MacOS using a Docker Desktop alternative (like Colima or Podman), Docker.DotNet still can't read environment variables; setting `$DOCKER_HOST` does nothing. Only symlink works to my knowledge.

```bash
sudo ln -sf $(docker context inspect --format '{{.Endpoints.docker.Host}}' | sed 's#^unix://##') /var/run/docker.sock
```
